### PR TITLE
Nick carousel top100

### PR DESCRIPTION
**Pull request by ngaffney22**
I still have to fix the JS with accessing the API I will have to meet with a TA because I don't understand it enough to iterate over the array of movie titles I created. For looks I temporarily put in images though so we still have a working image carousel. I just don't have any more time for this assignment today. they might allow a late submission tomorrow though so I can re-submit it.

### CODE REVIEW

**Review by KamranDjourshari24 — APPROVED** [review]
Looks good

**Comment by lavendermistyco** [comment]
Hey, I can't pull up your branch on my computer.

File: client/top-100/top-100Script.js [hunk 1]
```diff
@@ -53,6 +53,34 @@ document.addEventListener('DOMContentLoaded', () => {
     }));
     return top100moviesArray;
   }
-
-  imageExtractor(top100List);
-});
\ No newline at end of file
+  const movieList = ['Avengers:+Endgame', 'The+Dark+Knight', 'Django+Unchained', 'Spider-Man:+Into+the+Spider-Verse']
+  const imageList = [];
+  async function getImage() {
```
Comment (lavendermistyco) [inline client/top-100/top-100Script.js:58]: getImage(imageList)

File: client/top-100/top-100Script.js [hunk 2]
```diff
@@ -53,6 +53,34 @@ document.addEventListener('DOMContentLoaded', () => {
...
+    for (let i = 0; i < movieList.length; i++) {
+      const frontUrl = 'https://api.themoviedb.org/3/search/movie?api_key=ee671a8f0facbafb2b7da0decd7157db&query='
```
Comment (lavendermistyco) [inline client/top-100/top-100Script.js:60]: const imageList = [ ];

File: client/top-100/top-100Script.js [hunk 3]
```diff
@@ -53,6 +53,34 @@ document.addEventListener('DOMContentLoaded', () => {
...
+      const results = await fetch(frontUrl+`${movieList[i]}`);
+      const movieResults = await results.json();
+      console.log(movieResults.results[0].backdrop_path);
+      const backdropPath = movieResults.results[0].backdrop_path;
+      const imageRequest = await fetch(`https://image.tmdb.org/t/p/w342/${backdropPath}`);
+      const img = await imageRequest.blob();
+      imageList.push(img);
+      console.log(imageRequest);
+    }
```
Comment (lavendermistyco) [inline client/top-100/top-100Script.js:69]: return imageList;

File: client/top-100/top-100Script.js [hunk 4]
```diff
@@ -53,6 +53,34 @@ document.addEventListener('DOMContentLoaded', () => {
...
+  }
+  getImage()
```
Comment (lavendermistyco) [inline client/top-100/top-100Script.js:71]: const imageSlides = getImage(imageList);

File: client/top-100/top-100Script.js [hunk 5]
```diff
@@ -53,6 +53,34 @@ document.addEventListener('DOMContentLoaded', () => {
...
+  console.log(imageList);
+  const img1 = document.getElementsByName('slide1');
+  img1.src = imageList[0];
```
Comment (lavendermistyco) [inline client/top-100/top-100Script.js:74]: img1.src = imageSlides[0] ;

File: client/top-100/top-100Script.js (hunk 4 above)
Comment (lavendermistyco) [reply client/top-100/top-100Script.js:71]: This explains why it was null because it wasn't  returning any arraylist. Make sense? @ngaffney22

**Review by lavendermistyco — CHANGES_REQUESTED** [review]

File: client/top-100/top-100Script.js [hunk 6]
```diff
@@ -53,6 +53,34 @@ document.addEventListener('DOMContentLoaded', () => {
...
+    for (let i = 0; i < movieList.length; i++) {
+      const frontUrl = 'https://api.themoviedb.org/3/search/movie?api_key=ee671a8f0facbafb2b7da0decd7157db&query='
+      const results = await fetch(frontUrl+`${movieList[i]}`);
+      const movieResults = await results.json();
+      console.log(movieResults.results[0].backdrop_path);
+      const backdropPath = movieResults.results[0].backdrop_path;
+      const imageRequest = await fetch(`https://image.tmdb.org/t/p/w342/${backdropPath}`);
+      const img = await imageRequest.blob();
+      imageList.push(img);
+      console.log(imageRequest);
```
[the remark this answers is not in the document]
Comment (lavendermistyco) [reply client/top-100/top-100Script.js:68]: Instantiate a list here in this methos instead, so 
Const imageList = [ ];

File: client/top-100/top-100Script.js (hunk 1 above)
Comment (lavendermistyco) [inline client/top-100/top-100Script.js:58]: getImage(movieList);

**Review by lavendermistyco — CHANGES_REQUESTED** [review]

File: client/top-100/top-100Script.js [hunk 7]
```diff
@@ -53,6 +53,34 @@ document.addEventListener('DOMContentLoaded', () => {
     }));
     return top100moviesArray;
   }
-
-  imageExtractor(top100List);
-});
\ No newline at end of file
+  const movieList = ['Avengers:+Endgame', 'The+Dark+Knight', 'Django+Unchained', 'Spider-Man:+Into+the+Spider-Verse']
+  const imageList = [];
```
Comment (lavendermistyco) [inline client/top-100/top-100Script.js:57]: Place this line inside your async function

**Review by lavendermistyco — CHANGES_REQUESTED** [review]

**Pull request merged**